### PR TITLE
Improve TenantSelector component

### LINKED ISSUE #47: Add base-url option to TenantSelector
Our TenantSelector only allows you to select a tenant (project name) from the `https://api.cognitedata.com` api. But we also have other base urls like `https://*.cognitedata.com` where other tenants live. So maybe add a new prop to TenantSelector where you can enable a baseUrl field to override the base url.

### PR DESCRIPTION
**Pull request by mrtraser**
Expand functionality with ability to provide additional fields for component
Fixes #47

### CODE REVIEW

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-70.gearboxjs.preview.cogniteapp.com

File: src/components/TenantSelector/TenantSelector.tsx [hunk 1]
```diff
@@ -30,25 +39,65 @@ export interface TenantSelectorProps {
...
 
+  renderAdvancedOptions() {
+    const { advanced } = this.state;
+    const keys = Object.keys(advanced);
+
+    if (!keys.length) {
+      return null;
+    }
+
+    const inputs = Object.keys(advanced).map(option => {
```
Comment (f1cognite) [inline src/components/TenantSelector/TenantSelector.tsx]:
```suggestion
    const inputs = keys.map(option => {
```

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-70.gearboxjs.preview.cogniteapp.com

**Comment by evancharlton** [comment]
For future PRs: can you use better titles than "Improve TenantSelector"? That gives me (the reader) no idea what that change contains or why. "Edited code" is equally as informative.

Here's a good writeup of why good commit messages matter -- **especially** for open-source projects -- and how to write them: https://chris.beams.io/posts/git-commit/

**Comment by mrtraser** [comment]
Got it, I'll take a look on [how to write a git commit message](https://chris.beams.io/posts/git-commit/), thank you.

**Pull request merged**